### DISCUSSION TITLE
Using "from" header for alias email and "reply-to" header for reverse-alias (To mask the reverse-alias address)

### DISCUSSION THREAD

**Discussion by jumpingbrowntrout** [simple-login/app#546] [Feature Requests]
I noticed that another reputable alias service provider has a function to enable using "Use Reply-To Header For Replying"

Testing this and looking at the headers it shows the from address as the alias (instead of the reverse alias) but the "reply-to" header contains the reverse-alias and it's not visible to the user (in protonmail at least)

The end result is that the emails looks MUCH cleaner e.g 
from = alias@mydomain.com VS ra+alias.at.mydomain.com+stmpv@simplelogin.co

I would like to request this as a feature for a more aesthetically pleasing inbox experience.

**Comment by nguyenkims** [comment]
If the From: header to set to the alias, the email will seem to come from the alias and not sent to the alias right? I think this would be confusing.

Using the Reply-To trick doesn't work with multi-recipient emails or CC: is set.

**Reply by jumpingbrowntrout** [reply]
Good points @nguyenkims 
I forgot to take into consideration the multi-recipient / CC emails.
Let's drop this one :)

**Reply by rowatt** [reply]
Couldn't the from address just be the real, original sender address, and then the reply-to is the reverse-alias? That way, everything looks normal, except that the reply-to address ensures that replies are routed back properly.

I can see cases where that would not work - e.g. if there were other to/cc addresses which also needed routing through SimpleLogin - but having the reverse-alias in the reply-to would I think work for 99.9% of my use cases, so having that as an option would be great.

**Reply by nguyenkims** [reply]
@rowatt the From address can't be set to the real sender address as otherwise, the email will fail the DMARC.

**Reply by jumpingbrowntrout** [reply]
@rowatt I had that thought initially as well but also realized it would through DMARC off and likely your client would/could treat it as spam as you're essentially spoofing their address.

**Reply by rowatt** [reply]
Ah OK, that makes sense.

However, I used to do this myself with Mailgun forwarding and I never had any problems relating to DMARC etc. Mailgun wasn't adding/changing the reply-to header, but it was adding various x-headers... is forwarding through Mailgun doing something different, or does changing reply-to break whatever they are doing?

**Reply by jumpingbrowntrout** [reply]
Why don't you try dig up an old mail from mailgun where that was done and copy the headers here so we can take a look @rowatt

**Reply by rowatt** [reply]
Here you go:

```
Delivered-To: mailbox@mailbox.example.com
Received: by 2002:a17:907:75ec:0:0:0:0 with SMTP id jz12csp629869ejc;
        Thu, 1 Jul 2021 08:11:27 -0700 (PDT)
X-Google-Smtp-Source: ABdhPJyJDDFY+fKW3uqttz1wOLj0cwCKq/mzpiZJdAc/UdOoKs5hvKblS9DuEnJSikPNHfjvLnBN
X-Received: by 2002:aca:b443:: with SMTP id d64mr13370oif.68.1625152287482;
        Thu, 01 Jul 2021 08:11:27 -0700 (PDT)
ARC-Seal: i=1; a=rsa-sha256; t=1625152287; cv=none;
        d=google.com; s=arc-20160816;
        b=C9zNpS4/rKLStWne5XK0m+QDGyGtoQE/aWn6AJEjPBtquiONrMlyWY0HEPsy3b1qEg
         9fv1gAmLmvaVaQfVwxRi2ZB8uFa9UulvoYXYYEbuTChYG4B4bbpvPiDQt5T8rt4LO0NO
         pXDrz7OU+CWZvcz4vVqY9lQBjGFy/K6s31txbDzTfZPWj69LdciKbTnWlVYaTkcEt5km
         U3KeEjL8SE9rjpBG8gbf/OfWHdHl01ZGRv++jCvQCnm+YGO9yIXFgXvcZdCs8uy7P/vs
         t2NO14Mq7RF8ohxiWjxPjbvxXRmmNgIwGfBFzDJhDhFEGEha76skUYiAZXuPKM8vbgAq
         QRqA==
ARC-Message-Signature: i=1; a=rsa-sha256; c=relaxed/relaxed; d=google.com; s=arc-20160816;
        h=mime-version:subject:message-id:to:from:dkim-signature:date
         :dkim-signature;
        bh=pyQ8oD4e9sMNTd7RnQflx8gXD0P+w4lF1w6xM9kw59I=;
        b=ZoU5rjQUInd5lrFdQkbcBMK0UximrC6HjH4faebQuZmMXTHl0aATgORrnA9xDpQ3Vp
         zt9dg50RVTv4nG6qDxK0BqMgt2bSKtuDLqMK0uxIw1sbz7pBR1nN1g8B/BwTPFi9PovI
         U2dlsTBRf8OC/qPgZ4GCT7iZQExywfbHyzNE+UItJFoQUM2WYmi2N2Z8iauM7o0pROL+
         x1SP9T+GKIf35Rj5aN6jqNxC9/DK5/IuK6DrOwMVNvDOq3T1Ywd3p/p1uGdhbASaaA7K
         MfxPNFmFkGxD6b+oibRzIMpgmb+HJg08ERDr93wkeZCp5Gr6svwfrOrocC810T7Nmnx6
         5zcQ==
ARC-Authentication-Results: i=1; mx.google.com;
       dkim=pass header.i=@alias.example.com header.s=mx header.b=arhS7h4E;
       dkim=pass header.i=@linode.com header.s=k1 header.b="AaCBss/Q";
       spf=pass (google.com: domain of bounce+45017e.1b7c7f-mailbox=mailbox.example.com@alias.example.com designates 209.61.151.224 as permitted sender) smtp.mailfrom="bounce+45017e.1b7c7f-mailbox=mailbox.example.com@alias.example.com";
       dmarc=pass (p=REJECT sp=QUARANTINE dis=NONE) header.from=linode.com
Return-Path: <bounce+45017e.1b7c7f-mailbox=mailbox.example.com@alias.example.com>
Received: from rs224.mailgun.us (rs224.mailgun.us. [209.61.151.224])
        by mx.google.com with UTF8SMTPS id d1si343052oiw.24.2021.07.01.08.11.23
        for <mailbox@mailbox.example.com>
        (version=TLS1_3 cipher=TLS_AES_128_GCM_SHA256 bits=128/128);
        Thu, 01 Jul 2021 08:11:27 -0700 (PDT)
Received-SPF: pass (google.com: domain of bounce+45017e.1b7c7f-mailbox=mailbox.example.com@alias.example.com designates 209.61.151.224 as permitted sender) client-ip=209.61.151.224;
Authentication-Results: mx.google.com;
       dkim=pass header.i=@alias.example.com header.s=mx header.b=arhS7h4E;
       dkim=pass header.i=@linode.com header.s=k1 header.b="AaCBss/Q";
       spf=pass (google.com: domain of bounce+45017e.1b7c7f-mailbox=mailbox.example.com@alias.example.com designates 209.61.151.224 as permitted sender) smtp.mailfrom="bounce+45017e.1b7c7f-mailbox=mailbox.example.com@alias.example.com";
       dmarc=pass (p=REJECT sp=QUARANTINE dis=NONE) header.from=linode.com
DKIM-Signature: a=rsa-sha256; v=1; c=relaxed/relaxed; d=alias.example.com; q=dns/txt; s=mx; t=1625152287; h=Content-Type: MIME-Version: Subject: Message-ID: To: From: Date; bh=pyQ8oD4e9sMNTd7RnQflx8gXD0P+w4lF1w6xM9kw59I=; b=arhS7h4Ee3t3RCl5FhcCccxUUuTYm8BbkcQA86LqsWnTSUdUmV6doL5xQJOPrM5/M69LQEmU pv1n7kVg8s1ndkzbtFECSBp28DfjZuwjU4CDbnTtEEWI5z02uD6e2/h/kg7tC8z13e1riJ+P u4Evz3r0z7hoIevU0PvPxuxOWbo=
X-Mailgun-Sending-Ip: 209.61.151.224
X-Mailgun-Spam-Rules: DKIM_SIGNED, DKIM_VALID, DKIM_VALID_AU, DKIM_VALID_EF, HTML_MESSAGE, RCVD_IN_DNSWL_HI, RCVD_IN_MSPIKE_H4, RCVD_IN_MSPIKE_WL, SPF_HELO_NONE, SPF_PASS
X-Mailgun-Dkim-Check-Result: Pass
X-Mailgun-Spf: Pass
X-Mailgun-Sscore: -5.2
X-Mailgun-Sflag: No
X-Mailgun-Incoming: Yes
X-Envelope-From: <support@linode.com>
Received: from outbound-mail4.linode.com (outbound-mail4.linode.com [66.175.211.242]) by mxa.mailgun.org with ESMTP id 60dddb13.7f89ecc10830-api-n20; Thu, 01 Jul 2021 15:11:15 -0000 (UTC)
Date: Thu, 1 Jul 2021 11:11:14 -0400 (EDT)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/simple; d=linode.com; s=k1; t=1625152274; bh=6+V6Ic1JS4rXF/oSoRKrtl8ow+wfFSCwfvzWTAGMRsE=; h=Date:From:To:Subject:From; b=AaCBss/QQ+l8ICbFSkEqlkfDxB6OI4Pnur3jS4EgHcTR7SSk0fPmJt6zL1bX2lYaB
	 6KNW2HN5YZhGOqRGavtZcsc59MTPyE9nzcfQ2fZJ3+atz1glbR0yGPVqVuGu7RTDBm
	 +3vPygEIxdJvOPEO/nUXIMUKcDEiACEwdeTBOJiECDOoAhVabH22Iy07TZnXyWsvgu
	 dN1+s6ZIYlwqz9Bm1dx/kluMMK7VOIHVtnjrqs21fIkJ3rDmN8IgCQPFNaSu/PE71D
	 ajgWU4sa8NaEfgD20DKOu2a2tERs6i6FeNh/iQWR2zCKdQbp+miXyWkxnDm0wnmwiZ
	 hXRbxmfpzmqVA==
From: Linode Support <support@linode.com>
To: myalias@alias.example.com
Message-ID: <939019023.956527.1625152274348@outbound.mail.linode.com>
Subject: Meeting Linode's Customer Support Team
MIME-Version: 1.0
Content-Type: multipart/alternative; boundary="----=_Part_956526_1234071123.1625152274348"
X-Mailer: ColdFusion 11 Application Server
```

I changed the my email addresses/domains in the header. `mailbox.example.com` is the domain of the mailbox where emails were forwarded to, `alias.example.com` is the domain I use for aliases.

**Reply by jumpingbrowntrout** [reply]
@rowatt Mmmm interesting... Curious, what results do you get when you paste the original unaltered header into https://mxtoolbox.com/EmailHeaders.aspx
Does it show everything green/compliant ?
![image](https://user-images.githubusercontent.com/19516522/127349749-80450049-eadc-4f25-8811-3c3d27f270a4.png)

**Reply by rowatt** [reply]
Almost - SPF aligment is red, but the rest are green:
<img width="313" alt="Screenshot 2021-07-29 at 13 24 35" src="https://user-images.githubusercontent.com/339488/127491364-5e824121-f4ba-4d73-b6a6-551f4e980af6.png">

**Reply by jumpingbrowntrout** [reply]
So I just did a quick refresh of SPF Alignment here: https://mxtoolbox.com/dmarc/spf/spf-alignment?lm=NAV-UDC

_"To better understand how SPF alignment, let’s examine its makeup. Two types of alignment tests exist. One is based on SPF, and one is based on DKIM. The Alignment test for SPF is performed in order to verify the authenticity of the domain sending the email by using two header signatures in the message where the sender's domain is present. This is done as part of the DMARC standard, as the <From:> domain can be easily spoofed by fraudsters and used to trick people into opening malicious emails and clicking malicious links."_

From what I understand it checks "mail-from" domain against the "from" domain to see if it matches and if it does then SPF alignment passes.

So in your case (I obviously can't see the results since you need to mask your personal details :)) but if you click into the details and click more details it will take you to another page and so you exactly why it failed. The reason will be that linode.com does not match your mailfrom domain.

If we use your header as an example:

MailFROM
smtp.mailfrom="bounce+45017e.1b7c7f-mailbox=mailbox.example.com@alias.example.com";
From
Linode Support <support@linode.com>

It's interesting that this test still shows DMARC compliant when SPF alignment fails.

**Reply by rowatt** [reply]
Thanks for the explanation. Given my experience doing this with Mailgun, I don't think there were any negative consequences for me with SPF alignment failing. I suppose it's possible that it meant that a few more things were classified as spam, but in general all spam classification of mail routed through Mailgun worked fine (received to Google Workspace).

Is it possible that SimpleLogin could optionally forward in a similar way to Mailgun, leaving the sender email intact (but updating reply-to)?

**Reply by nguyenkims** [reply]
@rowatt ARC can be used to keep the sender address intact, you can follow its progress on  https://github.com/simple-login/app/discussions/504.
